src/Homeworks/Homework2_Task2/Trie.cs

```diff
@@ -0,0 +1,91 @@
+namespace Trees;
+public class Trie {
+    private readonly Node head;
+    public Trie() {
+        head = new Node();
+    }
+    public int Size => head.CountOfWords;
+    public bool Contains(string element) {
+        if (element == null) {
+            throw new ArgumentNullException("NULL");
+        }
+        Node currentnode = head;
+        foreach (var sign in element) {
+            if (!currentnode.Next.ContainsKey(sign)) {
+                return false;
+            }
+            currentnode = currentnode.Next[sign]; 
+        }
+        return currentnode.Terminal;
+    }
+    public bool Add(string element) {
+        if (element == null) {
+            throw new ArgumentNullException("Null");
+        }
+        if (element == string.Empty) {
+            return head.Terminal 
+            ? false
+            : head.Terminal = true;
+        }
+        if (Contains(element)) {
+            return false;
+        }
+        var currentnode = head;
+        foreach (var sign in element) {
+            currentnode.CountOfWords += 1;
+            if (!currentnode.Next.ContainsKey(sign)) {
+                currentnode.Next.Add(sign, new Node());
+            }
+            currentnode = currentnode.Next[sign];
+        }
+        currentnode.CountOfWords += 1;
+        currentnode.Terminal = true;
+        return currentnode.Terminal;
```

Comment:
Лучше просто ```return true;```, чтобы было читабельнее
```suggestion
        return currentnode.Terminal;
```